airbyte/sources/base.py

```diff
@@ -217,10 +222,14 @@ def validate_config(self, config: dict[str, Any] | None = None) -> None:
             )
             raise validation_ex from ex
 
-    def get_available_streams(self) -> list[str]:
+    def get_available_streams(self) -> list[str] | None:
         """Get the available streams from the spec."""
+        if not self._config and self.requires_config():
```

Comment:
@Suraj-Vishwakarma70 - I might be missing something but I don't see `requires_config()` declared in the code. Is it possible there's another change that didn't get pushed?

Reply:
So sorry about that @aaronsteers . 🙇 
I had made a couple of changes after that but it slipped out of my mind to push them.
I've pushed them and you can review them.